lib/services/ios-debug-service.ts

```diff
@@ -0,0 +1,294 @@
+	constructor(private $platformService: IPlatformService,
+		private $iOSEmulatorServices: Mobile.IEmulatorPlatformServices,
+		private $devicesServices: Mobile.IDevicesServices,
+		private $platformsData: IPlatformsData,
+		private $projectData: IProjectData,
+		private $childProcess: IChildProcess,
+		private $logger: ILogger,
+		private $fs: IFileSystem,
+		private $errors: IErrors,
+		private $injector: IInjector,
+		private $npm: INodePackageManager) {
+	}
+
+	get platform(): string {
+		return "ios";
+	}
+
+	public debug(): IFuture<void> {
+		if ((!options.debugBrk && !options.start) || (options.debugBrk && options.start)) {
+			this.$errors.failWithoutHelp("Expected exactly one of the --debug-brk or --start options.");
+		}
+
+		if (options.emulator) {
+			if (options.debugBrk) {
+				return this.emulatorDebugBrk();
+			} else if (options.start) {
+				return this.emulatorStart();
+			}
+		} else {
+			if (options.debugBrk) {
+				return this.deviceDebugBrk();
+			} else if (options.start) {
+				return this.deviceStart();
+			}
+		}
+
+		this.$errors.failWithoutHelp("Failed to select device or emulator to debug on.");
+	}
+
+	private emulatorDebugBrk(): IFuture<void> {
+		return (() => {
+			var emulatorDebugging = new iOSXcrunSimctl(this.$childProcess);
+			var device = emulatorDebugging.getRunningEmulator().wait();
+			if (device) {
+				this.$logger.info("Install and run on: " + device.uuid);
+			} else {
+				this.$errors.failWithoutHelp("This command expects a running emulator.");
+			}
+
+			var platformData = this.$platformsData.getPlatformData(this.platform);
+			this.$platformService.buildPlatform(this.platform).wait();
+			var emulatorPackage = this.$platformService.getLatestApplicationPackageForEmulator(platformData).wait();
+
+			device.installApp(emulatorPackage.packageName).wait();
+
+			this.executeOpenDebuggerClient().wait();
+
+			device.launchApp(this.$projectData.projectId, "--nativescript-debug-brk").wait();
+		}).future<void>()();
+	}
+
+	private emulatorStart(): IFuture<void> {
+		return (() => {
+			this.executeOpenDebuggerClient().wait();
+
+			var emulatorDebugging = new iOSXcrunSimctl(this.$childProcess);
+
+			var device = emulatorDebugging.getRunningEmulator().wait();
+			if (!device) {
+				this.$errors.failWithoutHelp("This command expects a running emulator.");
+			}
+
+			var projectId = this.$projectData.projectId;
+			var attachRequestMessage = notification.attachRequest(projectId);
+			device.postNotification(attachRequestMessage).wait();
+		}).future<void>()();
+	}
+
+	private deviceDebugBrk(): IFuture<void> {
+		return (() => {
+			this.$devicesServices.initialize({ platform: this.platform, deviceId: options.device }).wait();
+			this.$devicesServices.execute(device => (() => {
+				this.$platformService.deployOnDevice(this.platform).wait();
+				var deviceDebugging = new IOSDeviceDebugging(this.$projectData.projectId, <iOSDevice.IOSDevice>device, this.$logger, this.$injector);
```

Comment:
You can use `this.$injector.resolve(IOSDeviceDebuggins, { projectId: this.$projectData.projectId})`